django_file_form/views/s3multipart.py

```diff
@@ -0,0 +1,185 @@
+import json
+
+from django.http import JsonResponse
+from storages.utils import setting, lookup_env
+
+import boto3
+import os
+
+
+class s3multipart:
+    endpoint_url = setting('AWS_S3_ENDPOINT_URL')
+    aws_access_key_id = setting('AWS_S3_ACCESS_KEY_ID',
+                                setting('AWS_ACCESS_KEY_ID'))
+    aws_secret_access_key = setting('AWS_S3_SECRET_ACCESS_KEY',
+                                    setting('AWS_SECRET_ACCESS_KEY'))
+    aws_storage_bucket_name = setting('AWS_STORAGE_BUCKET_NAME')
+    file_form_upload_dir = setting('FILE_FORM_UPLOAD_DIR', 'temp_uploads')
+
+    @classmethod
+    def get_presigned_url(cls, request):
+        if request.method == 'POST':
+            # check if the post request has the file part
+            client = boto3.client(
+                's3',
+                endpoint_url=cls.endpoint_url or
+                lookup_env(['AWS_S3_ENDPOINT_URL', 'AWS_ENDPOINT_URL']),
+                aws_access_key_id=cls.aws_access_key_id or
+                lookup_env(['AWS_S3_ACCESS_KEY_ID', 'AWS_ACCESS_KEY_ID']),
+                aws_secret_access_key=cls.aws_secret_access_key or lookup_env(
+                    ['AWS_S3_SECRET_ACCESS_KEY', 'AWS_SECRET_ACCESS_KEY']))
+
+            json_body = json.loads(request.body)
+            fileName = json_body["filename"]
+            contentType = json_body["contentType"]
+            key = cls.file_form_upload_dir + "/" + fileName
```

Comment:
@jma7 Is it possible to add an unique ID between `FILE_FORM_UPLOAD_DIR` and `fileName`? Right now if multiple users upload to the same S3 account, the files will clash. It is better to add an per-form or per-user ID here. Perhaps we should even add another layer of ID and allow users to specify a directory in addition to `FILE_FORM_UPLOAD_DIR` (e.g. multiple forms inside an application). I suppose in this case the `direct_to_s3` variable in `View` could be used.

In this end we could have

```
BUCKET_NAME/UNIQUE_ID/USER_FOLDER/FILENAME
```